### PR TITLE
JP-1178: Updates for NRC_TSGRISM with NINTS=1

### PR DESCRIPTION
**Pull request by hbushouse**
While experimenting with the possibility of processing TSO exposures that have NINTS=1, I've determined that minor changes to a few level-2b (calwebb_tso-spec2) routines would be necessary to allow for this. The changes mainly involve backing off of the hardwired checks in many of the dedicated TSO routines that only allow for processing of a ``CubeModel``, as well as associated changes here and there to account for the fact that the data arrays are 2D instead of the expected 3D.

The changes made here to the ``photom`` step ``save_area_info`` function are actually needed for ALL modes of processing, because right now that function isn't getting called when there's no AREA reference file for the data. It does still need to be called, because as of recently it was updated to retrieve the pixel area keyword values from the PHOTOM ref file instead of the AREA ref file. So it can still be called even when there isn't an AREA ref file and in fact *must* be called for many such mode, because the pixel area keyword values are needed in downstream processing (e.g. ``extract_2d`` and/or ``extract_1d``).

### CODE REVIEW

File: jwst/extract_1d/extract.py [hunk 1]
```diff
@@ -2983,7 +2983,7 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
                     else:
                         wcs = input_model.meta.wcs
                     # There's no need to repeat this for each integration.
-                    if integ == 0:
+                    if integ <= 1:
```
Comment (hbushouse) [inline jwst/extract_1d/extract.py]: The reason for this change is due to the fact that in upstream code ``integrations`` is set to "[-1]" when the input data have a shape of 2 (as opposed to 3), so when executing this loop ``integ`` is equal to -1. The test for a value of zero was then causing this part of the if-block to get skipped, resulting the the pixel area values not getting populated. Now the pixel area is populated.

File: jwst/extract_2d/grisms.py [hunk 2]
```diff
@@ -189,9 +189,9 @@ def extract_tso_object(input_model,
         ymax = int(ymax)
 
         # cut it out, keep the entire row though
-        ext_data = input_model.data[:, ymin: ymax + 1, :].copy()
-        ext_err = input_model.err[:, ymin: ymax + 1, :].copy()
-        ext_dq = input_model.dq[:, ymin: ymax + 1, :].copy()
+        ext_data = input_model.data[..., ymin: ymax + 1, :].copy()
+        ext_err = input_model.err[..., ymin: ymax + 1, :].copy()
+        ext_dq = input_model.dq[..., ymin: ymax + 1, :].copy()
```
Comment (hbushouse) [inline jwst/extract_2d/grisms.py]: This change allows for both 2D and 3D array shapes.

File: jwst/photom/photom.py [hunk 3]
```diff
@@ -990,8 +991,8 @@ def apply_photom(self, photom_fname, area_fname):
 
         # Load the pixel area reference file, if it exists, and attach the
         # reference data to the science model
-        if area_fname != 'N/A':
-            self.save_area_info(ftab, area_fname)
+        #if area_fname != 'N/A':
+        self.save_area_info(ftab, area_fname)
```
Comment (hbushouse) [inline jwst/photom/photom.py]: Always call the ``save_area_info`` function and handle the absence of an AREA reference file within the function itself.

File: jwst/extract_1d/extract.py (hunk 1 above)
Comment (philhodge) [reply jwst/extract_1d/extract.py]: This is JP-1005, GitHub #4040.  However, this whole section will be deleted by addressing JP-1106, GitHub #4207.

**Comment by hbushouse** [comment]
Updated to conform to recent changes to extract_1d/extract.py (my change no longer needed). Going to merge in order to get the updates to photom installed, which are needed to fix other problems unrelated to the TSO NINTS=1 issue. The other changes to accommodate TSO NINTS=1 don't hurt anything else.

**Pull request merged**